src/ripple/core/LoadFeeTrack.h

```diff
@@ -39,37 +40,99 @@ namespace ripple {
 class LoadFeeTrack
 {
 public:
-    /** Create a new tracker.
-    */
-    static LoadFeeTrack* New (beast::Journal journal);
+    explicit LoadFeeTrack (beast::Journal journal = beast::Journal())
+        : m_journal (journal)
+        , mLocalTxnLoadFee (lftNormalFee)
+        , mRemoteTxnLoadFee (lftNormalFee)
+        , mClusterTxnLoadFee (lftNormalFee)
+        , raiseCount (0)
+    {
+    }
 
     virtual ~LoadFeeTrack () { }
 
     // Scale from fee units to millionths of a ripple
-    virtual std::uint64_t scaleFeeBase (std::uint64_t fee, std::uint64_t baseFee,
-                                        std::uint32_t referenceFeeUnits) = 0;
+    std::uint64_t scaleFeeBase (std::uint64_t fee, std::uint64_t baseFee,
+        std::uint32_t referenceFeeUnits) const;
 
     // Scale using load as well as base rate
-    virtual std::uint64_t scaleFeeLoad (std::uint64_t fee, std::uint64_t baseFee,
-                                        std::uint32_t referenceFeeUnits,
-                                        bool bAdmin) = 0;
-
-    virtual void setRemoteFee (std::uint32_t) = 0;
-
-    virtual std::uint32_t getRemoteFee () = 0;
-    virtual std::uint32_t getLocalFee () = 0;
-    virtual std::uint32_t getClusterFee () = 0;
-
-    virtual std::uint32_t getLoadBase () = 0;
-    virtual std::uint32_t getLoadFactor () = 0;
-
-    virtual Json::Value getJson (std::uint64_t baseFee, std::uint32_t referenceFeeUnits) = 0;
-
-    virtual void setClusterFee (std::uint32_t) = 0;
-    virtual bool raiseLocalFee () = 0;
-    virtual bool lowerLocalFee () = 0;
-    virtual bool isLoadedLocal () = 0;
-    virtual bool isLoadedCluster () = 0;
+    std::uint64_t scaleFeeLoad (std::uint64_t fee, std::uint64_t baseFee,
+        std::uint32_t referenceFeeUnits, bool bAdmin) const;
+
+    void setRemoteFee (std::uint32_t f)
+    {
+        ScopedLockType sl (mLock);
+        mRemoteTxnLoadFee = f;
+    }
+
+    std::uint32_t getRemoteFee () const
+    {
+        ScopedLockType sl (mLock);
+        return mRemoteTxnLoadFee;
+    }
+
+    std::uint32_t getLocalFee () const
+    {
+        ScopedLockType sl (mLock);
+        return mLocalTxnLoadFee;
+    }
+
+    std::uint32_t getClusterFee () const
+    {
+        ScopedLockType sl (mLock);
+        return mClusterTxnLoadFee;
+    }
+
+    std::uint32_t getLoadBase () const
+    {
+        return lftNormalFee;
+    }
+
+    std::uint32_t getLoadFactor () const
+    {
+        ScopedLockType sl (mLock);
+        return std::max(mClusterTxnLoadFee, std::max (mLocalTxnLoadFee, mRemoteTxnLoadFee));
```

Comment:
I only moved this code from another file. It looks like this is merely pre-C++11 code, because the two param version requires `algorithm`, too. I'll go ahead and made the change.